### ISSUE TITLE
Allow multiple media items to be uploaded for changelog entries

### ISSUE THREAD

**Issue by zacharlie** [kartoza/projecta#1219]
From the changelog entry editing interface, only a single media item can be uploaded and attached to the entry. For the QGIS changelog, more and more entries have additional media which is relevant to the entry, however only a single item may be uploaded at a time.

As a work around, each media element is currently attached independantly one at a time and references to the uploaded location are used within the body of the entry in order to include the additional media.

Whether each entry should be allowed multiple elements for "thumbnails" or whether a bulk uploader for media to use in the entry description is an architectural decision, but adding this functionality would reduce the workload on editors.

**Comment by sumandari** [comment]
@zacharlie 

could you please confirm if the idea is to modify input file in `/<project>/version/<version>/update/`from single input to multiple input looks like demo:

![imageedit_1_5146705965](https://user-images.githubusercontent.com/40058076/96108355-0c964480-0f10-11eb-969d-4cfd6de8adba.gif)


how it should be displayed on the detail view page? will it be using boostrap corousel?

**Comment by zacharlie** [comment]
Yes, the general idea is what you have depicted. I don't know the specifics on how the implementation will affect the outputs which get used in https://qgis.org/en/site/forusers/visualchangelogs.html, so maybe @rduivenvoorde or @timlinux might have a suggestion as to how the want the media to be exposed/ stored/ used, but on the detail page within projecta I imagine just having multiple thumbnails which are displayed in a single column (underneath each other) should be a suitable arrangement.

I think a way to reference these images  (e.g. fig-1, fig-2) may be very useful as well for my use case, but that implementation likely warrants further discussion.

![image](https://user-images.githubusercontent.com/64078329/96120926-9550b880-0eef-11eb-8caf-8b8d0ce53f8e.png)

**Comment by rduivenvoorde** [comment]
Although we are talking about a 'visible changelog', I would stick to one figure per item. We already have animated gifs/screenies, and I think that should be enough. IF people really want to talk so much about their feature, let us point them to the documentation writing and let them write a huge section there. That is more helpfull that writing huge feature items :-)

Also because (well to me) the main purpose is still to create simple valid rst. The more images we put in (which are being copied AND multiplied to the number of translations(!) the website is getting bigger and bigger....

**Comment by zacharlie** [comment]
Noted. Initially when I had started this process of changelog maintenance I aimed for minimal interference (from the maintenance side), so that I don't make work for others if I fall off the earth, but it certainly seems that removing verbose content from the changelog and leaving the extra media for the documentation team is a good approach.

I still feel that the ability to add multiple media items would be useful for certain instances, especially when a single PR performs multiple functions (e.g. 3.16: Add options to dynamically tweak dash pattern in simple line symbol layers). At this point the media is just getting embedded in the entry body.

Certainly the duplication for each translation is an issue which needs to be resolved...

**Comment by sumandari** [comment]
Hi, @zacharlie 

It took time for me to get back here for learning the code. Please correct me if I were wrong, there's an image_file in Entry model. From my humble opinion, we can create a new model to manage additional images. If user uploads more than one image, the rest will be saved in additional models. I'm not sure if the order in row needs to be a concern for the images.
I'm looking forward having further discussion/suggestion for this idea.

**Comment by zacharlie** [comment]
I think just being able to include multiple images should be enough for now. If anybody needs further functionality or a change in the implementation we can perhaps open a new ticket.
